CONTRIBUTING.md

```diff
@@ -129,6 +129,40 @@ Run `rush build -t <packagename>` to build a single project, and all local proje
 
 By default, Rush only displays things written to `STDERR`. If you want to see the full output, pass `--verbose` to any of the build commands. This is also true for script commands (see below).
 
+### Testing
+
+Once the project has successfully built, you can test any of our packages by executing `rushx test` from the root of the package that you want to test. This will automatically run the tests in both NodeJS and in a browser. To target these environments individually, you can run `rushx test:node` and `rushx test:browser`.
+
+By default, these npm scripts run recorded tests. We use a custom recording library called `test-utils-recorder`. You can learn more about our recorder at the [test-utils-recorder's readme](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/test-utils/recorder/README.md).
+
+#### Recorded tests
+
+To run unit tests, go to the root of the project you want to test and run `rushx unit-test`. These tests aim to reproduce the behavior of our library against remote endpoints through previously recorded HTTP request and responses. You won't need to have any environment configuration, nor internet access, for these tests to run.
+
+Keep in mind that if you modify the test files or the underlying network calls of the project you're testing, the recordings will need to be re-generated.
+
+You can read more in the section: [Regenerating recordings](#regenerating-recordings).
+
+#### Live tests
+
+By default, the npm scripts `test`, `test:node` or `test:browser` run recorded tests. To run live tests, set the environment variable `TEST_MODE` to "live" and ensure you have the right Azure resources to test against.
+
+Our integration tests will run against the live resources, which are determined by the environment variables you provide.
+
+To run the integration tests, you will need to execute `rushx integration-test`.
+
+To generate new resources for your tests, follow the procedure at [Integration Testing with live services](https://github.com/Azure/azure-sdk-for-js/blob/master/CONTRIBUTING.md#integration-testing-with-live-services). What resources get created before the test, or created during the test depends on how the ARM template named `test-resources.json` is built, just as much as how the tests are written. Generally speaking, the ARM templates will create the main account in which individual tests will create the resources they test. For example, for the Service Bus client, the [ARM template](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/test-resources.json) only creates the Service Bus namespace (with the appropriate configurations), while the tests automatically create the queues, topics and subscriptions.
```

Comment:
> To generate new resources for your tests, follow the procedure at [Integration Testing with live services](...)

Any reason to not combine the contents of the `Integration Testing with live services` section here?

> Generally speaking, the ARM templates will create the main account in which individual tests will create the resources they test. For example, for the Service Bus client, the [ARM template](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/servicebus/test-resources.json) only creates the Service Bus namespace (with the appropriate configurations), while the tests automatically create the queues, topics and subscriptions.

Do we have to get into the details here? I would expect the individual test readmes to talk about what the corresponding ARM template would create.

Reply:
Gotcha, I thought you would like this since it's an attempt on reusing stuff. I'll make sure each readme has information about what each ARM template generates.

Reply:
Done!